Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -1269,8 +1262,12 @@ private static void CleanupCheckers(string requestId)
 
       if (CommandLineOptions.Clo.XmlSink != null)
       {
-        CommandLineOptions.Clo.XmlSink.WriteEndMethod(verificationResult.Outcome.ToString().ToLowerInvariant(),
-          verificationResult.End, verificationResult.End - verificationResult.Start);
+        lock (CommandLineOptions.Clo.XmlSink) {
```

Comment:
I'm pretty sure this was the start of an attempt to make `/xml` thread safe, but I would revert it and keep this PR focussed on just adding the resource count, and punt on #460 for now

Reply:
Done.